release/rancher/rancher.go

```diff
@@ -515,19 +514,59 @@ func formatContentLine(line string) string {
 	return strings.TrimSpace(line)
 }
 
-func GenerateMissingImagesList(imagesListURL, registry, username, password string, concurrencyLimit int, checkImages, ignoreImages []string, verbose bool) ([]string, error) {
-	if len(checkImages) == 0 {
-		if imagesListURL == "" {
-			return nil, errors.New("if no images are provided, an images list URL must be provided")
+// ImagesLocationsMap searches for missing images in a registry and creates a map with the locations of the images, or if they are missing
+// this map can be used to identify where which image should be synced from
+func ImagesLocationsMap(username, password string, concurrencyLimit int, checkImages, ignoreImages []string, targetRegistry string, imagesRegiestries []string) (map[string][]string, error) {
+	imagesLocations := make(map[string][]string)
+
+	var err error
```

Comment:
Why is this necessary?

Reply:
It was in a previous version, not anymore, removed